src/components/ToolBar/AppMenu/menu-factory.tsx

```diff
@@ -427,27 +429,80 @@ export const AppMenuItemDialog: React.FC<AppMenuItemProps> = (props) => {
     </Box>
   )
 
-  return (
-    <Dialog
-      maxWidth="sm"
-      fullWidth
-      open={open}
-      onClose={handleClose}
-      onKeyDown={(e) => {
-        e.stopPropagation()
-      }}
-      onClick={(e) => {
-        e.stopPropagation()
-      }}
-    >
-      <Box sx={{ p: 3.5 }}>
-        <Typography variant="h5">{app.name}</Typography>
+return (
+  <Dialog
+    maxWidth="sm"
+    fullWidth
+    open={open}
+    onClose={handleClose}
+    onKeyDown={(e) => {
+      e.stopPropagation();
+    }}
+    onClick={(e) => {
+      e.stopPropagation();
+    }}
+  >
+    <Box sx={{ p: 3.5 }}>
+      <Stack spacing={1.5}>
+        <Typography variant="h5">
+          {app.name}
+        </Typography>
+
+        <Typography variant="body1">
+  {app.showDescriptionInDialog ? (app.description || "No description available.") : null}
+</Typography>
+
         {inputDefinition}
-        {parametersSection}
-        {submitButton}
-      </Box>
-    </Dialog>
-  )
+
+        {/* ↓↓↓ MODIFIED parametersSection rendering below ↓↓↓ */}
+        {app.parameters && (
+  <Box sx={{ p: 3 }}>
+    {/* Hide "parameters" header */}
+    {app.parameters.map((parameter: ServiceAppParameter) => {
+      const isSpecialCheckbox =
+        app.name?.toLowerCase() === "networkx analyzer";
```

Comment:
Should remove this special case. If this is a show stopper could add logic where this special checkbox is put in if an app only has one parameter and it is a checkbox. Please check with Jing to see if that makes sense.